### PR TITLE
Only use CoordsXY objects in world/Footpath.cpp

### PR DESCRIPTION
**Pull request by tupaschoal**
I double check and it seems like this patch eliminates all the remaining `{x, y}` usages on footpath.

### CODE REVIEW

File: src/openrct2/world/Footpath.cpp [hunk 1]
```diff
@@ -1678,18 +1669,12 @@ static TileElement* footpath_can_be_wide(int32_t x, int32_t y, uint8_t height)
...
+void footpath_update_path_wide_flags(const CoordsXY& footpathPos)
 {
-    if (x < 0x20)
-        return;
-    if (y < 0x20)
-        return;
-    if (x > 0x1FDF)
-        return;
-    if (y > 0x1FDF)
+    if (footpathPos.x < 0x20 || footpathPos.y < 0x20 || footpathPos.x > 0x1FDF || footpathPos.y > 0x1FDF)
```
Comment (Gymnasiast) [inline src/openrct2/world/Footpath.cpp]:
```suggestion
    if (map_is_location_at_edge(footpathPos))
```

File: src/openrct2/world/Footpath.cpp [hunk 2]
```diff
@@ -1731,24 +1716,23 @@ void footpath_update_path_wide_flags(int32_t x, int32_t y)
...
-        x += 0x20;
-        pathList[4] = footpath_can_be_wide(x, y, height);
-        y -= 0x20;
-        pathList[5] = footpath_can_be_wide(x, y, height);
-        y -= 0x20;
-        pathList[6] = footpath_can_be_wide(x, y, height);
-        x -= 0x20;
-        pathList[7] = footpath_can_be_wide(x, y, height);
-        y += 0x20;
+        auto pathPos = footpathPos - CoordsXY{ 0x20, 0x20 };
```
Comment (Gymnasiast) [inline src/openrct2/world/Footpath.cpp]:
```suggestion
        auto pathPos = footpathPos - CoordsXY{ COORDS_XY_STEP, COORDS_XY_STEP };    
```

Also applies to the rest of the `0x20`s.

File: src/openrct2/world/Footpath.cpp (hunk 1 above)
Comment (tupaschoal) [reply src/openrct2/world/Footpath.cpp]: Great reminder, thanks!

**Comment by tupaschoal** [comment]
Fixed pending issues.

File: src/openrct2/world/Footpath.cpp [hunk 3]
```diff
@@ -1731,24 +1716,23 @@ void footpath_update_path_wide_flags(int32_t x, int32_t y)
...
-        pathList[4] = footpath_can_be_wide(x, y, height);
-        y -= 0x20;
-        pathList[5] = footpath_can_be_wide(x, y, height);
-        y -= 0x20;
-        pathList[6] = footpath_can_be_wide(x, y, height);
-        x -= 0x20;
-        pathList[7] = footpath_can_be_wide(x, y, height);
-        y += 0x20;
+        auto pathPos = footpathPos - CoordsXY{ COORDS_XY_STEP, COORDS_XY_STEP };
+        pathList[0] = footpath_can_be_wide(pathPos, height);
```
Comment (duncanspumpkin) [inline src/openrct2/world/Footpath.cpp]: Could you put a todo saying use DirectionDelta

Comment (tupaschoal) [reply src/openrct2/world/Footpath.cpp]: Done

**Pull request merged**